transcom/mymove
# B 18309 multi moves landing page component creation

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A879850&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

As we transition to allow for multiple moves, a new home page and component must be created to display to the customer. 

Right now in order to see this component, we are using a feature flag in `featureFlags.js` called `multiMove` that is being implemented in [this Agility Ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A873911&RoomContext=TeamRoom%3A808731&concept=TeamRoom) and [this PR](https://github.com/transcom/mymove/pull/11753). I have merged that PR into this PR in order to use that functionality.

This PR does the following:
- New component `MultiMovesLandingPage` that is the parent component holding containers of moves/shipments for current & previous moves
- tests for `MultiMovesLandingPage`
- New component `MultiMovesMoveHeader` that takes in a title and displays a moving truck icon to the left of the title
- tests for `MultiMovesMoveHeader`
- New component `MultiMovesMoveContainer` that holds data pertaining to current & previous moves. This is a reusable component that maps over an array of objects containing move data
- tests for `MultiMovesMoveContainer`
- New component `MultiMovesInfoList` that shows a grid/table of pertinent move data that can be seen at a glance. There is conditional logic that does determine different displays based on `orderType`
- tests for `MultiMovesInfoList`
- New component `ButtonDropdownMenu` that takes in an array of objects and displays a dropdown menu that is web/mobile friendly and uses a function passed in as a parameter. Also takes in parameter to determine if it is outlined or not and various classes/styles as needed
- tests for `ButtonDropdownMenu`
- test data to be used in the above mentioned components to mimic upcoming API call to be created and used on user load


### How to test

1. Access MM as a customer and sign in
2. Navigate to `http://milmovelocal:3000/moves` to see the new dashboard
3. This is currently using test data so it should display the same for everyone

## Screenshots
Expanded
<img width="794" alt="Screenshot 2024-01-22 at 11 58 08 AM" src="https://github.com/transcom/mymove/assets/136510600/64a05e9e-ef38-4280-ad27-a2c991c4d98f">

Condensed
<img width="1708" alt="Screenshot 2024-01-22 at 11 57 55 AM" src="https://github.com/transcom/mymove/assets/136510600/8d3f33a2-83be-41f6-832d-6782375e01c8">
